src/components/workspace/core/floating-window.tsx

```diff
@@ -0,0 +1,239 @@
+export const FloatingWindow = memo(({ windowId, window, onSnapRequest, children }: FloatingWindowProps) => {
+    const dispatch = useDispatch();
+    const intl = useIntl();
+    const [dragMousePos, setDragMousePos] = useState<{ x: number; y: number } | null>(null);
+    const [isDragging, setIsDragging] = useState(false);
+
+    const displayTitle = intl.messages[window.title] ? intl.formatMessage({ id: window.title }) : window.title || '';
+
+    const handleDrag = useCallback((_e: any, _d: { x: number; y: number }) => {
+        setIsDragging(true);
+        const event = globalThis.window.event as MouseEvent | undefined;
+        if (event) {
+            setDragMousePos({ x: event.clientX, y: event.clientY });
+        }
+    }, []);
+
+    const handleDragStop = useCallback(
+        (_e: any, data: { x: number; y: number }) => {
+            setIsDragging(false);
+            setDragMousePos(null);
+            dispatch(updateWindowPosition({ windowId, position: { x: data.x, y: data.y } }));
+        },
+        [dispatch, windowId]
+    );
+
+    const handleResizeStop = useCallback(
+        (_e: any, _direction: any, ref: HTMLElement, _delta: any, position: { x: number; y: number }) => {
+            dispatch(
+                updateWindowSize({
+                    windowId,
+                    size: { width: parseInt(ref.style.width, 10), height: parseInt(ref.style.height, 10) },
+                })
+            );
+            dispatch(updateWindowPosition({ windowId, position }));
+        },
+        [dispatch, windowId]
+    );
+
+    const handleFocus = useCallback(() => {
+        if (!isDragging) {
+            dispatch(focusWindow(windowId));
+        }
+    }, [dispatch, windowId, isDragging]);
+
+    useEffect(() => {
+        if (dragMousePos && !window.isMaximized && !window.isPinned) {
+            onSnapRequest(windowId, dragMousePos);
+        }
+    }, [dragMousePos, windowId, onSnapRequest, window.isMaximized, window.isPinned]);
+
+    if (window.isMinimized) {
+        return null;
+    }
+
+    return (
+        <Rnd
+            default={{
+                x: window.position.x,
+                y: window.position.y,
+                width: window.size.width,
+                height: window.size.height,
+            }}
+            position={{
+                x: window.position.x,
+                y: window.position.y,
+            }}
+            size={{
+                width: window.isMaximized ? '100%' : window.size.width,
+                height: window.isMaximized ? '100%' : window.size.height,
+            }}
+            onDrag={handleDrag}
+            onDragStop={handleDragStop}
+            onResizeStop={handleResizeStop}
+            dragHandleClassName="window-header"
+            cancel=".window-content"
+            disableDragging={window.isMaximized || window.isPinned}
+            enableResizing={!window.isMaximized && !window.isPinned}
+            minWidth={300}
+            minHeight={200}
```

Comment:
use minSize defined in workspace.constants.ts instead ? or remove it from default config